sources/packages/api/src/services/institution/institution.service.ts

```diff
@@ -52,7 +95,70 @@ export class InstitutionService extends RecordDataModelService<Institution> {
     institution.primaryEmail = institutionDto.primaryEmail;
     institution.website = institutionDto.website;
     institution.regulatingBody = institutionDto.regulatingBody;
-    institution.establishedDate = new Date(institutionDto.establishedDate);
+    institution.establishedDate = institutionDto.establishedDate;
+
+    //Institution Primary Contact Information
+    institution.institutionPrimaryContact = {
+      primaryContactFirstName: institutionDto.primaryContactFirstName,
+      primaryContactLastName: institutionDto.primaryContactLastName,
+      primaryContactEmail: institutionDto.primaryContactEmail,
+      primaryContactPhone: institutionDto.primaryContactPhone,
+    };
+
+    //Institution Legal Authority Contact Information
+    institution.legalAuthorityContact = {
+      legalAuthorityFirstName: institutionDto.legalAuthorityFirstName,
+      legalAuthorityLastName: institutionDto.legalAuthorityLastName,
+      legalAuthorityEmail: institutionDto.legalAuthorityEmail,
+      legalAuthorityPhone: institutionDto.legalAuthorityPhone,
+    };
+
+    //Institution Address
+    institution.institutionAddress = {
+      addressLine1: institutionDto.addressLine1,
+      addressLine2: institutionDto.addressLine2,
+      city: institutionDto.city,
+      provinceState: institutionDto.provinceState,
+      country: institutionDto.country,
+      postalCode: institutionDto.postalCode,
+      phone: institutionDto.primaryPhone,
+    };
+
+    return await this.save(institution);
+  }
+
+  async updateInstitution(userInfo: UserInfo, institutionDto: InstituteDto) {
+    const account = await this.bceidService.getAccountDetails(
+      userInfo.idp_user_name,
+    );
+
+    if (account == null) {
+      //This scenario occurs if basic BCeID users try to push the bceid account into our application.
```

Comment:
This scenario should no longer be valid because we are blocking Basic BCeID to authenticate.
We can keep the code but maybe change the comment.